apps/src/lab2/views/components/Instructions.tsx

```diff
@@ -106,8 +115,12 @@ interface InstructionsPanelProps {
   messageIndex?: number;
   /** Optional image URL to display. */
   imageUrl?: string;
-  /** If the next button should be shown. */
-  showNextButton?: boolean;
+  /** If the continue button should be shown. */
+  showContinueButton?: boolean;
+  /** If the finish button should be shown. */
+  showFinishButton?: boolean;
+  /** Additional callback to fire before navigating to the next level. */
+  beforeNextLevel?: () => void;
```

Comment:
nit: maybe better to call this `beforeFinish` or something to more clearly indicate what this callback is used for? Technically this inner component doesn't know that `beforeNextLevel` happens to already be wrapped by the `onNextPanel` callback

Reply:
That makes sense - thanks!